libs/SalesforceSDK/src/com/salesforce/androidsdk/auth/idp/IDPManager.kt

```diff
@@ -98,24 +107,38 @@ internal class IDPManager(
             }
 
             override fun generateAuthCode(
-                context: Context,
+                authCodeActivity: IDPAuthCodeActivity,
                 userAccount: UserAccount,
                 spConfig: SPConfig,
                 codeChallenge: String,
                 onResult: (result: IDPAuthCodeHelper.Result) -> Unit
             ) {
-                IDPAuthCodeHelper.generateAuthCode(WebView(context), userAccount, spConfig, codeChallenge, onResult)
+                IDPAuthCodeHelper.generateAuthCode(
```

Comment:
Previously we were building the web view right here. Now we pull it from the auth code activity.
NB: the method does not need an actual activity simply an object implementing the IDPAuthCodeActivity. That makes testing easier.